osu.Game/Rulesets/Scoring/JudgementProcessor.cs

```diff
@@ -134,6 +133,10 @@ void simulate(HitObject obj)
             }
         }
 
-        private void updateHasCompleted() => hasCompleted.Value = JudgedHits == MaxHits;
+        protected override void Update()
+        {
+            base.Update();
+            hasCompleted.Value = JudgedHits == MaxHits && (JudgedHits == 0 || lastAppliedResult.TimeAbsolute < Clock.CurrentTime);
```

Comment:
Personally I'm okay with this change, but would like a second opinion from @smoogipoo (see https://github.com/ppy/osu/pull/12376/commits/d0f30b7b422afe385fd1fc0de7a28684bf0ab7ef for reasoning).

Reply:
Though it's super dodgy, I think I'm fine with this. I'd like this to not be a thing in the long term though.